Fixes to get kore 4.1.0 building in Ubuntu

osfameron opened the pull request:
This now builds in Docker (e.g. Ubuntu) with the same version 4.1.0 that's built on OSX (though without pgsql support) but spews following error on starting in dev mode:

```
root@e9e0490aaf3c:/try-cb-lcb# kodev run
building try-cb-lcb (dev)
CFLAGS=-std=c11 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -Wall -Wpedantic -Wextra -Wshadow -fPIC -Isrc -Isrc/includes -I/usr/local/include  -g -DDEBUG
LDFLAGS=-lcouchbase -ljwt -luuid -shared
nothing to be done!
[parent]: notls serving http on 127.0.0.1:8080
[parent]: privsep: no root path set, using working directory
[parent]: privsep: will not change user
[parent]: privsep: no keymgr_root set, using 'root` directory
[parent]: privsep: will not chroot
[parent]: kore is starting up
[parent]: seccomp sandbox enabled
[parent]: Couchbase Connection: couchbase://db
[parent]: Couchbase Username: Administrator
[wrk 4]: worker 4 started (cpu#4, pid#5030)
[parent]: worker [wrk 4] (5030) exited with status 31
[parent]: worker 4 (pid: 5030) (hdlr: none) gone
[parent]: worker 4 died from sandbox violation
[parent]: restarting worker 4
...
```

Running the prod script instead fails with
`kore: configuration error on line 4`

osfameron committed:
Fix Dockerfile running on Ubuntu Linux

 * remove `apt-get` install of kore, as it is outdated version
 * install as per https://docs.kore.io/4.1.0/install.html
 * Kore under Linux filters syscalls using seccomp
   * see https://docs.kore.io/4.1.0/api/seccomp.html
   * this then shuts down the worker processes as soon as libcouchbase makes a
     disallowed syscall
   * added tracing `seccomp_tracing yes` to config, to enable logging of the
     offending syscalls
   * declare those calls in the C source
 * Add `set -x` to wait-for-couchbase for better logging of what command is
   executed
 * Change bound IP address to 0.0.0.0, which is required when connecting from
   outside the Docker container (e.g. from web browser running on the host)

osfameron commented:
> If you are running on Linux, Kore will be using seccomp to disallow system calls that it does not require.

This seems to be the reason for the "sandbox violation". I can't see a mechanism to just Switch This Feature Off, but it has fine control over which syscalls are allowed. Presumably one of the calls to Couchbase is triggering the violation.

Kore doesn't seem to log *which* syscall triggers it, otherwise https://docs.kore.io/4.1.0/api/seccomp.html should be able to deal with this?

(I've compiled with DEBUG=1, and I believe Kore sends output to syslog, but `dmesg` doesn't seem to work within docker, so haven't found anything useful.)

@raycardillo tl;dr, this seems to be a Linux + Kore.io issue where the libcouchbase calls are not allowed by Kore's SECCOMP filter. I don't have the C/unix knowledge to debug this further without help.

osfameron commented:
OK, updated with all fixes. This now builds and runs for me with `docker-compose up`.

@raycardillo: I've not re-tested running locally on Mac, I'll have a look at that in due course.

The installation of `kore` is a bit clumsy. I think `make install` should actually, well, *install* it, but I had to `cp` the files myself. Any suggestions there welcome, but not critical.

raycardillo committed:
Fixes for OSX

maria-robobug commented on Dockerfile:
```diff
@@ -29,10 +23,30 @@ RUN apt-get install -y libcouchbase3 \
 
 # Set the entrypoint
-ENTRYPOINT ["./wait-for-couchbase.sh", "kore", "-n", "-r", "-c", "/try-cb-lcb/conf/try-cb-lcb.conf"]
+# ENTRYPOINT ["./wait-for-couchbase.sh", "kodev", "run"]
```
Probably can get rid of the commented out line now?

raycardillo replied:
@osfameron @maria-robobug  More background... If using the `kore` command, I added the `-f` to the other invocation example (run in foreground) for easier debugging since a syslogger is not installed in the instance. Also need the `-r` and `-c` unless you plan to get all that special setup going as well (but that's overkill for this purpose). However, given the intent of this, perhaps it is easier to just use the `kodev run` command instead. I have not heartburn either way ultimately. ;)

maria-robobug commented on README.md:
````diff
@@ -142,7 +142,7 @@ See [Kore.io] documentation for more details on various ways to run the server.
 
 ```
-CB_HOST=127.0.0.1 CB_USER=raycardillo CB_PSWD=raycardillo ./dev-run.sh
+CB_HOST=127.0.0.1 CB_USER=raycardillo CB_PSWD=raycardillo ./run-dev.sh
````
Not for this line, but for the README in general, since Docker works now (🥳 ) we can get rid of the warnings. For example:
https://github.com/couchbaselabs/try-cb-lcb/pull/3/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R59

maria-robobug reviewed (approved):
👍 looks good on my side, just minor comments to address

raycardillo commented on src/try-cb-lcb.c:
```diff
@@ -23,6 +23,27 @@
+// syscalls required by libcouchbase
+// See https://docs.kore.io/4.1.0/api/seccomp.html
+// List of syscalls retrieved by running with `seccomp_tracing yes` in config.
+KORE_SECCOMP_FILTER("try-cb-lcb",
```
@osfameron When I ran through a quick sanity test, just realized we're missing one:
```
try-cb-api               | [parent]: seccomp violation, [wrk 2] pid=288, syscall=96:gettimeofday
```

osfameron replied:
Good catch, thanks! You must have tested a few more cases than I tried :)
(Or possibly the failure was non-fatal with tracing enabled, so I just missed it? Anyway, adding in.)

raycardillo reviewed (changes requested):
Need to add `gettimeofday` to the allow-list as well. Otherwise, looking good.

osfameron committed:
Remove warnings about Docker status

osfameron committed:
Add allowed syscall, simplify `kodev run`

As this is not a production deployment, it's simpler to showcase `kodev run`
in the Dockerfile.
